src/main/webapp/app/shared/layouts/navbar/navbar.component.ts

```diff
@@ -126,4 +154,25 @@ export class NavbarComponent implements OnInit, OnDestroy {
             }
         }
     }
+
+    /**
+     * check if exam already started
+     */
+    examHasStarted(exam?: Exam): boolean {
+        return exam?.startDate ? exam.startDate.isBefore(this.serverDateService.now()) : false;
+    }
+
+    /**
+     * check if exam is over
+     */
+    examIsOver(exam?: Exam): boolean {
+        return exam?.endDate ? exam.endDate.isBefore(this.serverDateService.now()) : true;
+    }
+
+    /**
+     * check if exam mode is active
+     */
+    examModeActive(exam?: Exam): boolean {
```

Comment:
Instead of passing it to each method you could just check `this.exam` since it is the object which you pass to them, otherwise the code looks good to me 👍

Reply:
done 😊